src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,250 @@
+
+    chunk_size = Int(
+        2500,
+        help="Size of the chunk used for the calculation of the statistical values",
+    ).tag(config=True)
+
+    def __call__(
+        self,
+        dl1_table,
+        masked_pixels_of_sample=None,
+        chunk_shift=None,
+        col_name="image",
+    ) -> list:
+        """
+        Prepare the extraction chunks and call the relevant function of the particular extractor
+        to extract the statistical values.
+
+        Parameters
+        ----------
+        dl1_table : astropy.table.Table
+            DL1 table with images of shape (n_images, n_channels, n_pix)
+            and timestamps of shape (n_images, ) stored in an astropy Table
+        masked_pixels_of_sample : ndarray
+            boolean array of masked pixels of shape (n_pix, ) that are not available for processing
+        chunk_shift : int
+            number of samples to shift the extraction chunk
+        col_name : string
+            column name in the DL1 table
+
+        Returns
+        -------
+        List StatisticsContainer:
+            List of extracted statistics and extraction chunks
+        """
+
+        # Check if the statistics of the dl1 table is sufficient to extract at least one chunk.
+        if len(dl1_table) < self.chunk_size:
+            raise ValueError(
+                f"The length of the provided DL1 table ({len(dl1_table)}) is insufficient to meet the required statistics for a single extraction chunk of size ({self.chunk_size})."
+            )
+
+        # Function to split the dl1 table into appropriated chunks
+        def _get_chunks(dl1_table, chunk_shift):
+            # Calculate the range step: Use chunk_shift if provided, otherwise use chunk_size
+            step = chunk_shift if chunk_shift is not None else self.chunk_size
+
+            # Generate chunks that do not overflow
+            for i in range(0, len(dl1_table), step):
+                if i + self.chunk_size <= len(dl1_table):
+                    yield dl1_table[i : i + self.chunk_size]
+
+            # If chunk_shift is None, ensure the last chunk is of size chunk_size, if needed
+            if chunk_shift is None and len(dl1_table) % self.chunk_size != 0:
+                yield dl1_table[-self.chunk_size :]
+
+        # Get the chunks of the dl1 table
+        dl1_chunks = _get_chunks(dl1_table, chunk_shift)
+
+        # Calculate the statistics from a chunk of images
+        stats_list = [
+            self.extract(
+                chunk[col_name].data, chunk["time_mono"], masked_pixels_of_sample
+            )
+            for chunk in dl1_chunks
+        ]
+
+        return stats_list
+
+    @abstractmethod
+    def extract(self, images, times, masked_pixels_of_sample) -> StatisticsContainer:
+        pass
+
+
+class PlainExtractor(StatisticsExtractor):
+    """
+    Extract the statistics from a chunk of peak time images
+    using numpy and scipy functions
+    """
+
+    time_median_cut_outliers = List(
```

Comment:
Is it necessary to define the outliers here in this extractor class?

The outliers are defined from information that is also stored in the output. Outliers can be defined in many different ways.

I think I'd prefer if the `StatsExtractor` class only did that: extract statistics.

The definitions of outliers in the extracted statistics should be done by something else, maybe an `OutlierDetector`

Reply:
Sounds reasonable to us. The outlier detection is remove from the PR and will be treated outside the stats extraction.